Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2328,34 +2333,81 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
+                // SymbolKit has already cloned the overload group symbol from an existing overload. The
+                // symbol identifier it gave the symbol starts with the cloned symbol's ID, followed by
+                // the fixed string '::OverloadGroup'. Since we want the cloned overload info to be
+                // first in the display list, scan through the list of symbol IDs and swap indices if
+                // necessary.
+                if let clonedIdIndex = overloadSymbolIds.firstIndex(where: { $0 + SymbolGraph.Symbol.overloadGroupIdentifierSuffix == overloadGroupId }) {
+                    overloadSymbolIds.swapAt(clonedIdIndex, overloadSymbolIds.startIndex)
+                }
+            }
 
+            func addOverloadReferences(
+                to documentationNode: DocumentationNode,
+                at index: Int,
+                overloadSymbolReferences: [DocumentationNode]
+            ) throws {
                 guard let symbol = documentationNode.semantic as? Symbol else {
                     preconditionFailure("""
-                    Only symbols can be overloads. Found non-symbol overload for \(symbolReference.absoluteString.singleQuoted).
-                    Non-symbols should already have been filtered out in `PathHierarchyBasedLinkResolver.traverseOverloadedSymbols(_:)`.
+                    Only symbols can be overloads. Found non-symbol overload for \(documentationNode.reference.absoluteString.singleQuoted).
+                    Non-symbols should already have been filtered out by SymbolKit.
                     """)
                 }
-                guard symbolReference.sourceLanguage == .swift else {
+                guard documentationNode.reference.sourceLanguage == .swift else {
                     assertionFailure("Overload groups is only supported for Swift symbols.")
-                    continue
+                    return
                 }
 
-                var otherOverloadedSymbolReferences = overloadedSymbolReferences
+                var otherOverloadedSymbolReferences = overloadSymbolReferences.map(\.reference)
                 otherOverloadedSymbolReferences.remove(at: index)
-                let overloads = Symbol.Overloads(references: otherOverloadedSymbolReferences, displayIndex: index)
+                let displayIndex: Int
+                if let symbolIndex = symbol.overloadsVariants.firstValue?.displayIndex {
+                    displayIndex = symbolIndex
+                } else {
+                    displayIndex = index
+                }
+                let overloads = Symbol.Overloads(references: otherOverloadedSymbolReferences, displayIndex: displayIndex)
                 symbol.overloadsVariants = .init(swiftVariant: overloads)
             }
+
+            try addOverloadReferences(to: overloadGroupNode, at: 0, overloadSymbolReferences: overloadSymbolNodes)
+
+            for (index, node) in overloadSymbolNodes.indexed() {
```

Comment:
Is it correct to start counting from `0` here if the overload group is already added at index `0` above? 

Should this be `zip(1..., overloadSymbolNodes)` instead?

Reply:
The overload group is a separate symbol from the symbol at index `0` - the latter is the original symbol that the overload group data was cloned from.

Reply:
@QuietMisdreavus and I chatted briefly about this because I was also confused--since the overload group is a duplicate of the first symbol in the list, they should use the same index so the group page can disguise itself properly. Thanks for the clarifying comment @QuietMisdreavus!